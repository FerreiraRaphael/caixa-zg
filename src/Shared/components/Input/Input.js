import React from "react";
import PropTypes from "prop-types";
import {
  Input,
  FormControl,
  InputLabel,
  IconButton,
  CircularProgress
} from "material-ui";
import { Search } from "material-ui-icons";

const CommonInputPropTypes = {
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  fullWidth: PropTypes.bool,
  autoFocus: PropTypes.bool,
  loading: PropTypes.bool
};

const CommonInputDefaultProps = {
  fullWidth: false,
  autoFocus: false,
  loading: false
};

class CommonInput extends React.Component {
  componentDidMount() {
    const { focus } = this.props;
    console.log("didmount", focus);
    if (focus) {
      this.input.focus();
    }
  }

  componentWillReceiveProps(newProps) {
    const { focus } = newProps;
    console.log("newprops", focus);
    if (focus) {
      this.input.focus();
    }
  }

  render() {
    const {
      label,
      fullWidth,
      autoFocus,
      loading,
      onChange,
      onBlur,
      onFocus,
      value
    } = this.props;
    return (
      <FormControl fullWidth={fullWidth}>
        <InputLabel>{label}</InputLabel>
        <Input
          onBlur={onBlur}
          onFocus={onFocus}
          inputRef={input => {
            this.input = input;
          }}
          autoFocus={autoFocus}
          endAdornment={
            <IconButton>
              {loading ? (
                <CircularProgress tabIndex={-1} />
              ) : (
                <Search tabIndex={-1} />
              )}
            </IconButton>
          }
          value={value}
          onChange={onChange}
        />
      </FormControl>
    );
  }
}
CommonInput.propTypes = CommonInputPropTypes;

CommonInput.defaultProps = CommonInputDefaultProps;

export default CommonInput;
