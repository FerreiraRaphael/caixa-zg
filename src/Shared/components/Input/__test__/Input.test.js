import React from "react";
import { shallow } from "enzyme";

import Input from "../Input";

describe("Input Component", () => {
  const setup = propsOverride => {
    const props = {
      onChange: jest.fn(),
      label: "test",
      value: "value",
      ...propsOverride
    };

    const wrapper = shallow(<Input {...props} />);

    return {
      props: wrapper.props(),
      wrapper
    };
  };

  it("should render without loading", () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with loading", () => {
    const { wrapper } = setup({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });
});
