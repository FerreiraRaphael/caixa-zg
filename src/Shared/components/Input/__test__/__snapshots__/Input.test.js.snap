// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render with loading 1`] = `
<WithStyles(FormControl)
  fullWidth={false}
>
  <WithStyles(InputLabel)>
    test
  </WithStyles(InputLabel)>
  <WithStyles(Input)
    autoFocus={false}
    endAdornment={
      <WithStyles(IconButton)>
        <WithStyles(CircularProgress)
          tabIndex={-1}
        />
      </WithStyles(IconButton)>
    }
    inputRef={[Function]}
    onChange={[Function]}
    value="value"
  />
</WithStyles(FormControl)>
`;

exports[`Input Component should render without loading 1`] = `
<WithStyles(FormControl)
  fullWidth={false}
>
  <WithStyles(InputLabel)>
    test
  </WithStyles(InputLabel)>
  <WithStyles(Input)
    autoFocus={false}
    endAdornment={
      <WithStyles(IconButton)>
        <pure(Search)
          tabIndex={-1}
        />
      </WithStyles(IconButton)>
    }
    inputRef={[Function]}
    onChange={[Function]}
    value="value"
  />
</WithStyles(FormControl)>
`;
